Wikipedia/Code/PreviewAndSaveViewController.m

```diff
@@ -197,6 +205,15 @@ - (void)goForward {
     }
 }
 
+- (void)goToArticleFor:(NSString *)href {
```

Comment:
I thought about this but I don't understand how adding the method to the delegate will prevent the manual construction of the URL. Is it possible for you to do this part? That way, I can look at the code and learn how to do it too.

Will update this Pull Request with a fix for the previous comment.